packages/next/src/compiled/react-dom/cjs/react-dom-server.edge.production.js

```diff
@@ -11,9 +11,9 @@
 'use strict';
 
 var React = require("next/dist/compiled/react");
-var ReactDOM = require('react-dom');
+var ReactDOM = require("next/dist/compiled/react-dom");
```

Comment:
The previous import source was incorrect since it resulted in multiple versions of react-dom being bundled. I
- [ ] double check all changed imports
- [ ] Check with why we previously didn't do this on purpose
- [ ] check if we can better guard in React or frameworks against multiple versions of React/ReactDOM (it's not just related to our issues but also React Hooks not working: https://github.com/facebook/react/issues/13991)